### ISSUE TITLE
qt5-base[latest]:x64-windows - single configuration builds of Qt (eg. release) are currently not supported

### ISSUE THREAD

**Issue by heydojo** [microsoft/vcpkg#10586]
**Host Environment**
- OS: x64-windows
- Compiler: Visual Studio 2019 Developer Command Prompt v16.5.0

**To Reproduce**

Set build types to release.
`echo.set(VCPKG_BUILD_TYPE release)>> triplets\x64-windows.cmake`
Install build Qt.
`vcpkg install qt5-base[latest]:x64-windows`
Build your Qt based C++ application with something like:
`mkdir build && cd build`
cmake .. -DQt5_DIR:PATH="%VCPKGDIR%\packages\qt5-base_x64-windows\share\cmake\Qt5"

**Failure logs** 
```
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.
CMake Error at windows/vcpkg/packages/qt5-base_x64-windows/share/cmake/Qt5Gui/Qt5GuiConfig.cmake:15 (message):
  The imported target "Qt5::Gui" references the file

     "H:/qt5/windows/vcpkg/packages/qt5-base_x64-windows/debug/bin/Qt5Guid.dll"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "H:/qt5/windows/vcpkg/packages/qt5-base_x64-windows/share/cmake/Qt5Gui/Qt5GuiConfig.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  windows/vcpkg/packages/qt5-base_x64-windows/share/cmake/Qt5Gui/Qt5GuiConfig.cmake:37 (_qt5_Gui_check_file_exists)
  windows/vcpkg/packages/qt5-base_x64-windows/share/cmake/Qt5Gui/Qt5GuiConfig.cmake:177 (_populate_Gui_target_properties)
  windows/vcpkg/packages/qt5-base_x64-windows/share/cmake/Qt5Widgets/Qt5WidgetsConfig.cmake:115 (find_package)
  windows/vcpkg/packages/qt5-base_x64-windows/share/cmake/Qt5/Qt5Config.cmake:28 (find_package)
  CMakeLists.txt:16 (find_package)


-- Configuring incomplete, errors occurred!
See also "H:/qt5/build/CMakeFiles/CMakeOutput.log".
```

Specifically:
```
The imported target "Qt5::Gui" references the file

     "H:/qt5/windows/vcpkg/packages/qt5-base_x64-windows/debug/bin/Qt5Guid.dll"

  but this file does not exist.
```

**Additional context**

Please fix or revert the regression.
https://github.com/microsoft/vcpkg/commit/f8165f72709cce797058164b389a105ca4c412a9#diff-f322c6a00cca5779dfc2edc84a72c0d7

https://github.com/microsoft/vcpkg/blob/master/ports/qt5-base/fixcmake.py#L30

Thanks.

**Comment by Neumann-A** [comment]
In the mentioned commit I corrected the configs to correctly include release and debug libraries if build with vcpkg's defaults. (The configs where incorrect before) 
I currently don't know if this is fully fixed in #9705 (or #9860 or #10009) (since I don't test single configuration builds) but I added in a few `if(EXISTS "someqtlib")` to the target properties population of the Qt targets. So it should be at least a bit better ;).
I also completely removed `fixcmake.py` in #9705 because the required fixes were to complicated to achieve when the configs were already generated

**Comment by heydojo** [comment]
@Neumann-A thanks for the reply. I tried to pull your feature branch at https://github.com/microsoft/vcpkg/pull/9705 and it unfortunately failed to build for me. It looks like it is ready to merge but one of the patches fails. I added a comment to the discussion, along with the console output.
Would you mind taking a look please?

**Comment by heydojo** [comment]
Just adding that I tested https://github.com/microsoft/vcpkg/pull/10644 and this issue is solved once it is merged into master.
